### DISCUSSION TITLE
Possible incorrect behaviour using notlike and an empty/void entry

### DISCUSSION THREAD

**Discussion by LexiconGap** [sqlalchemy/sqlalchemy#10794] [Usage Questions] [answered]
### Describe the bug

the result of a notlike operation on an empty/void column entry always returns False. Should it not always return True?
Like, everything is not in an empty set?

### SQLAlchemy Version in Use

1.4.39

### DBAPI (i.e. the database driver)

none

### Database Vendor and Major Version

sqlite

### Python Version

3.10

### Operating system

linux

### To Reproduce

```python
model_attr.notlike('any_value')
```

### Error

No error, just a behaviour

### Additional context

I'm a complete newbie so this may be normal behaviour

**Comment by zzzeek** [comment, marked as answer]
NULL does not mean "nothing" in relational databases.  it means "unknown".  so the result of any comparison operation with NULL will also be NULL.

```
>>> import sqlite3
>>> conn = sqlite3.connect(":memory:")
>>> cursor = conn.cursor()
>>> cursor.execute("select not NULL like 'some value'")
<sqlite3.Cursor object at 0x7f36e86ae4c0>
>>> cursor.fetchone()
(None,)
>>> cursor.execute("select NULL not like 'some value'")
<sqlite3.Cursor object at 0x7f36e86ae4c0>
>>> cursor.fetchone()
(None,)
>>> cursor.execute("select NULL like 'some value'")
<sqlite3.Cursor object at 0x7f36e86ae4c0>
>>> cursor.fetchone()
(None,)


```

**Comment by LexiconGap** [comment]
Thank you for the clarification
